api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -36,6 +36,54 @@
 The Tensors are rank 1 if ``batch_size`` is 1, otherwise the Tensors are rank 2."""
 
 
+class Encoder(abc.ABC):
+    """Base class for obs encoders.
+    To define a custom encoder, two methods must be implemented:
+
+    - ``register``: defines how the encoder will be fitted to the data.
+    - ``transform``: defines how the encoder will be applied to the data
+      in order to create an obs_tensor.
+
+    See the implementation of ``DefaultEncoder`` for an example.
+    """
+
+    @abc.abstractmethod
+    def register(self, obs: pd.DataFrame) -> None:
+        """Register the encoder with obs."""
+        pass
+
+    @abc.abstractmethod
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Transform the obs DataFrame into a DataFrame of encoded values."""
+        pass
+
+    @property
+    def name(self) -> str:
+        return self.__class__.__name__
+
+
+class DefaultEncoder(Encoder):
+    """Default encoder based on LabelEncoder."""
+
+    def __init__(self, col: str) -> None:
+        self._encoder = LabelEncoder()
+        self.col = col
+
+    def register(self, obs: pd.DataFrame) -> None:
+        self._encoder.fit(obs[self.col].unique())
+
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        return self._encoder.transform(df[self.col])  # type: ignore
+
+    @property
+    def name(self) -> str:
+        return self.col
+
+    @property
+    def classes_(self):  # type: ignore
```

Comment:
Worth pulling up into `Encoder` base class?

Reply:
I am not sure if this is something that necessarily applies to all encoders - I think we can keep this scoped to our DefaultEncoder.